tests/specs/int-simplification.k

```diff
@@ -0,0 +1,190 @@
+
+    rule (S1 +Int C2) +Int C3 => S1 +Int (C2 +Int C3) [concrete(C2, C3), symbolic(S1), simplification]
+    rule (S1 +Int C2) -Int C3 => S1 +Int (C2 -Int C3) [concrete(C2, C3), symbolic(S1), simplification]
+    rule (S1 -Int C2) +Int C3 => S1 +Int (C3 -Int C2) [concrete(C2, C3), symbolic(S1), simplification]
+    rule (S1 -Int C2) -Int C3 => S1 -Int (C2 +Int C3) [concrete(C2, C3), symbolic(S1), simplification]
+
+  // ###########################################################################
+  // inequality
+  // ###########################################################################
+
+    rule I1 +Int C2  <Int C3         => I1          <Int C3 -Int C2 [concrete(C2, C3), simplification]
+    rule C1          <Int I2 +Int C3 => C1 -Int C3  <Int I2         [concrete(C1, C3), simplification]
+    rule C1         <=Int I2 +Int C3 => C1 -Int C3 <=Int I2         [concrete(C1, C3), simplification]
+
+    rule X +Int Y <Int Z => X <Int Z -Int Y [concrete(Y), simplification]
+
+endmodule
+
+module INT-SIMPLIFICATION-COMMON
+    imports INT
+    imports BOOL
+
+  // ###########################################################################
+  // add, sub
+  // ###########################################################################
+
+  // 2 terms
+    rule A -Int A => 0 [simplification]
+    rule A -Int 0 => A [simplification]
+    rule 0 +Int A => A [simplification]
+    rule A +Int 0 => A [simplification]
+
+  // 3 terms
+    rule  (A -Int  B) +Int B  => A [simplification]
+    rule   A -Int (A  -Int B) => B [simplification]
+    rule   A +Int (B  -Int A) => B [simplification]
+    rule  (A +Int  B) -Int A  => B [simplification]
+
+  // 4 terms
+    rule  (A +Int B) +Int (C  -Int A) => B +Int C [simplification]
+    rule  (A +Int B) -Int (A  +Int C) => B -Int C [simplification]
+    rule  (A +Int B) -Int (C  +Int A) => B -Int C [simplification]
+    rule  (A +Int B) +Int (C  -Int B) => A +Int C [simplification]
+    rule ((A -Int B) -Int  C) +Int B  => A -Int C [simplification]
+
+  // 5 terms
+    rule   (A +Int  B  +Int C)  -Int (A  +Int D) =>  B +Int (C  -Int D) [simplification]
+    rule   (C +Int (A  -Int D)) +Int (B  -Int A) =>  C +Int (B  -Int D) [simplification]
+    rule (((A -Int  B) -Int C)  -Int  D) +Int B  => (A -Int  C) -Int D  [simplification]
+
+  // >=6 terms
+    rule (A +Int B +Int C +Int D +Int E) -Int (A +Int C +Int F) => B +Int D +Int E -Int F [simplification]
+
+  // ###########################################################################    
+  // mul
+  // ###########################################################################    
+
+    rule 1 *Int A => A
+    rule A *Int 1 => A
+    rule 0 *Int _ => 0
+    rule _ *Int 0 => 0
+
+    rule C *Int A +Int B *Int A => (C +Int B) *Int A [simplification]
+
+    rule E *Int A +Int B +Int C +Int D +Int F *Int A => (E +Int F) *Int A +Int B +Int C +Int D [simplification]
+
```

Comment:
I know that `*Int` is tighter binding, but can we put parens here to make it easier to read?